theforeman/foreman_virt_who_configure
# Fixes #37204 - Move proxy params to correct location in virt-who.conf

#### What are the changes introduced in this pull request?
* Fixed an issue where the HTTP PROXY dropdown was showing the full URL instead of name like other places in Foreman causing the UI to look odd
* Moved the `no_proxy` under `[system_environment]`
* Looking at the man page you only need to specify no_proxy if you want to ignore proxies and it should not be there by default so fixed the logic that was causing it to be there even if the field was empty and updated the tests
* moved `http/s_proxy` under `[system_environment]`

#### What are the testing steps for this pull request?
* Check out PR
* Create an http and https proxy, does not have to be valid just use a fake url
* Create and deploy a config without any proxy info and make sure `no_proxy` is not in the `/etc/virt-who.conf` config
* Edit that config and add * into the `ignore_proxy` UI field, save and deploy the config check and see if you see `no_proxy=*` in the `/etc/virt-who.conf` config
* Edit that config and chose an HTTP Proxy from the dropdown, save and deploy the config and make sure you see `http_proxy` under `[system_environment]`
* Edit that config and chose an HTTPS Proxy from the dropdown, save and deploy the config and make sure you see `https_proxy` under `[system_environment]`

#### If you don't want to deploy the config, you can just look at the script and see what it would put in `/etc/virt-who.conf` that is what I did when testing the changes before I tried to deploy them.

With my PR it should look like this for each scenario:

HTTPS Proxy:
```bash
### This configuration file is managed via the virt-who configure plugin
### manual edits will be deleted.
[global]
debug=1
interval=86400
oneshot=False

[system_environment]
https_proxy=https://admin:changeme@proxy.satellite.lab.eng.rdu2.redhat.com:3128
```

HTTP Proxy:
```bash
### This configuration file is managed via the virt-who configure plugin
### manual edits will be deleted.
[global]
debug=1
interval=86400
oneshot=False

[system_environment]
http_proxy=http://admin:changeme@proxy.satellite.lab.eng.rdu2.redhat.com:3128
```

Ignore Proxy:
```bash
### This configuration file is managed via the virt-who configure plugin
### manual edits will be deleted.
[global]
debug=1
interval=86400
oneshot=False

[system_environment]
no_proxy=*
```

No Proxy:
```bash
### This configuration file is managed via the virt-who configure plugin
### manual edits will be deleted.
[global]
debug=1
interval=86400
oneshot=False

[system_environment]
```
